apps/ops/models/job.py

```diff
@@ -188,11 +189,17 @@ def summary(self):
 
     @property
     def average_time_cost(self):
-        total_cost = 0
-        finished_count = self.executions.filter(status__in=['success', 'failed']).count()
-        for execution in self.executions.filter(status__in=['success', 'failed']).all():
-            total_cost += execution.time_cost
-        return total_cost / finished_count if finished_count else 0
+        executions = self.executions.filter(status__in=['success', 'failed'])
+        if not executions.exists():
+            return 0
+        average = executions.annotate(
+            time_cost=Coalesce(F('date_finished'), Now()) - F('date_start')
+        ).annotate(
+            time_cost_seconds=Extract('time_cost', 'epoch')
+        ).aggregate(
+            avg_time=Avg('time_cost_seconds')
+        )['avg_time']
+        return average if average is not None else 0
 
     def get_register_task(self):
         from ..tasks import run_ops_job
```

Comment:
The code looks fine. There are no apparent irregularities, potential issues, nor optimization suggestions required at this moment. However, it's always good practice to review the code regularly and update as necessary.